EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/attributeInput/AttributeInput.java

```diff
@@ -0,0 +1,306 @@
+	 * Will be overridden in subclasses.
+	 *
+	 * @return a default value as string
+	 */
+	String getDefaultInput() { return "UNDEFINED"; }
+
+	/**
+	 * Checks whether the user input is contained in a given range.
+	 * Might get overridden in subclass.
+	 *
+	 * @param range to be searched in
+	 * @return bool if string is in range
+	 */
+	public boolean isInRange(String range) {
+		String[] possibleValues = range.split(";");
+		for (String possibleValue : possibleValues) {
+			if (possibleValue.equals(this.getValueAsString())) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Returns the object as a string containing the most important information concerning the instance.
+	 *
+	 * @return a string describing the AttributeInput-Object
+	 */
+	@Override
+	public String toString() {
+		String lineSeparator = System.getProperty("line.separator");
+		StringBuilder stringify = new StringBuilder(lineSeparator);
+		stringify.append(this.getClass() + lineSeparator);
+		stringify.append("Current input: " + this.getInput() + lineSeparator);
+		stringify.append("Default input: " + this.getDefaultInput() + lineSeparator);
+		stringify.append("Input used: " + this.getInputOrDefault() + lineSeparator);
+		stringify.append("Current value: " + this.getValueAsString());
+		return stringify.toString();
+	}
+
+	/**
+	 * Checks whether the AttributeInput-Instance provides different methods that can be selected for computation.
+	 *
+	 * @return a bool indicating if different methods are supported
+	 */
+	public boolean hasDifferentMethods() {
+		return !this.getAvailableMethods().isEmpty();
+	}
+
+	/**
+	 * Returns a validator for wicket input fields fitting the AttributeInput and the selected method.
+	 *
+	 * @return an IValidator object from the generator.validation package
+	 */
+	public IValidator<String> getAttributeInputValidator() {
+		return AttributeValidator.getValidatorForAttribute(this.getAttribute());
+	}
+
+	/**
+	 * Get a random index for the provided array.
+	 *
+	 * @param inputArray the index should be generated for
+	 * @return an int containing the index
+	 */
+	public static int getRandomIndex(Object[] inputArray) {
+		return random.nextInt(inputArray.length);
+	}
+
+	/**
+	 *  Get a random index for the provided list.
+	 *
+	 * @param inputList the index should be generated for
+	 * @return an int containing the index
+	 */
+	public static int getRandomIndex(List inputList) { return random.nextInt(inputList.size()); }
+
+	/**
+	 * Enum containing all possible methods for value generation of any AttributeInput-Subclasses.
+	 */
+	public enum ProbabilityDistributionEnum implements Serializable {
+
+		UNIFORM("uniform"), NORMAL("normal");
+
+		private String distribution;
+
+		/**
+		 * Constructor for the enum, sets the given distribution string.
+		 *
+		 * @param distribution string to be set as distribution name
+		 */
+		ProbabilityDistributionEnum(final String distribution) {
+			this.distribution = distribution;
```

Comment:
What if I hand a random string here, which is not one of "uniform" and "normal"?